// code-theme-converter/dist/types/idea.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var IdeaColorNames;
(function (IdeaColorNames) {
    IdeaColorNames["ADDED_LINES_COLOR"] = "ADDED_LINES_COLOR";
    IdeaColorNames["ANNOTATIONS_COLOR"] = "ANNOTATIONS_COLOR";
    IdeaColorNames["CARET_COLOR"] = "CARET_COLOR";
    IdeaColorNames["CONSOLE_BACKGROUND_KEY"] = "CONSOLE_BACKGROUND_KEY";
    IdeaColorNames["DELETED_LINES_COLOR"] = "DELETED_LINES_COLOR";
    IdeaColorNames["DIFF_SEPARATORS_BACKGROUND"] = "DIFF_SEPARATORS_BACKGROUND";
    IdeaColorNames["DOCUMENTATION_COLOR"] = "DOCUMENTATION_COLOR";
    IdeaColorNames["ERROR_HINT"] = "ERROR_HINT";
    IdeaColorNames["FILESTATUS_ADDED"] = "FILESTATUS_ADDED";
    IdeaColorNames["FILESTATUS_COPIED"] = "FILESTATUS_COPIED";
    IdeaColorNames["FILESTATUS_DELETED"] = "FILESTATUS_DELETED";
    IdeaColorNames["FILESTATUS_IDEA_FILESTATUS_DELETED_FROM_FILE_SYSTEM"] = "FILESTATUS_IDEA_FILESTATUS_DELETED_FROM_FILE_SYSTEM";
    IdeaColorNames["FILESTATUS_IDEA_FILESTATUS_IGNORED"] = "FILESTATUS_IDEA_FILESTATUS_IGNORED";
    IdeaColorNames["FILESTATUS_MERGED"] = "FILESTATUS_MERGED";
    IdeaColorNames["FILESTATUS_MODIFIED"] = "FILESTATUS_MODIFIED";
    IdeaColorNames["FILESTATUS_RENAMED"] = "FILESTATUS_RENAMED";
    IdeaColorNames["FILESTATUS_UNKNOWN"] = "FILESTATUS_UNKNOWN";
    IdeaColorNames["FILESTATUS_ADDEDOUTSIDE"] = "FILESTATUS_ADDEDOUTSIDE";
    IdeaColorNames["FOLDED_TEXT_BORDER_COLOR"] = "FOLDED_TEXT_BORDER_COLOR";
    IdeaColorNames["GUTTER_BACKGROUND"] = "GUTTER_BACKGROUND";
    IdeaColorNames["INDENT_GUIDE"] = "INDENT_GUIDE";
    IdeaColorNames["INFORMATION_HINT"] = "INFORMATION_HINT";
    IdeaColorNames["LINE_NUMBERS_COLOR"] = "LINE_NUMBERS_COLOR";
    IdeaColorNames["LINE_NUMBER_ON_CARET_ROW_COLOR"] = "LINE_NUMBER_ON_CARET_ROW_COLOR";
    IdeaColorNames["METHOD_SEPARATORS_COLOR"] = "METHOD_SEPARATORS_COLOR";
    IdeaColorNames["MODIFIED_LINES_COLOR"] = "MODIFIED_LINES_COLOR";
    IdeaColorNames["NOTIFICATION_BACKGROUND"] = "NOTIFICATION_BACKGROUND";
    IdeaColorNames["QUESTION_HINT"] = "QUESTION_HINT";
    IdeaColorNames["RECENT_LOCATIONS_SELECTION"] = "RECENT_LOCATIONS_SELECTION";
    IdeaColorNames["RECURSIVE_CALL_ATTRIBUTES"] = "RECURSIVE_CALL_ATTRIBUTES";
    IdeaColorNames["RIGHT_MARGIN_COLOR"] = "RIGHT_MARGIN_COLOR";
    IdeaColorNames["SELECTED_INDENT_GUIDE"] = "SELECTED_INDENT_GUIDE";
    IdeaColorNames["SELECTED_TEARLINE_COLOR"] = "SELECTED_TEARLINE_COLOR";
    IdeaColorNames["SEPARATOR_BELOW_COLOR"] = "SEPARATOR_BELOW_COLOR";
    IdeaColorNames["SOFT_WRAP_SIGN_COLOR"] = "SOFT_WRAP_SIGN_COLOR";
    IdeaColorNames["TEARLINE_COLOR"] = "TEARLINE_COLOR";
    IdeaColorNames["VCS_ANNOTATIONS_COLOR_1"] = "VCS_ANNOTATIONS_COLOR_1";
    IdeaColorNames["VCS_ANNOTATIONS_COLOR_2"] = "VCS_ANNOTATIONS_COLOR_2";
    IdeaColorNames["VCS_ANNOTATIONS_COLOR_3"] = "VCS_ANNOTATIONS_COLOR_3";
    IdeaColorNames["VCS_ANNOTATIONS_COLOR_4"] = "VCS_ANNOTATIONS_COLOR_4";
    IdeaColorNames["VCS_ANNOTATIONS_COLOR_5"] = "VCS_ANNOTATIONS_COLOR_5";
    IdeaColorNames["VISUAL_INDENT_GUIDE"] = "VISUAL_INDENT_GUIDE";
    IdeaColorNames["WHITESPACES_MODIFIED_LINES_COLOR"] = "WHITESPACES_MODIFIED_LINES_COLOR";
    IdeaColorNames["WHITESPACES"] = "WHITESPACES";
})(IdeaColorNames = exports.IdeaColorNames || (exports.IdeaColorNames = {}));
var IdeaStyleTypeNames;
(function (IdeaStyleTypeNames) {
    IdeaStyleTypeNames["FONT_TYPE"] = "FONT_TYPE";
    IdeaStyleTypeNames["EFFECT_TYPE"] = "EFFECT_TYPE";
})(IdeaStyleTypeNames = exports.IdeaStyleTypeNames || (exports.IdeaStyleTypeNames = {}));
var IdeaAttributeColorNames;
(function (IdeaAttributeColorNames) {
    IdeaAttributeColorNames["FOREGROUND"] = "FOREGROUND";
    IdeaAttributeColorNames["BACKGROUND"] = "BACKGROUND";
    IdeaAttributeColorNames["ERROR_STRIPE_COLOR"] = "ERROR_STRIPE_COLOR";
    IdeaAttributeColorNames["EFFECT_COLOR"] = "EFFECT_COLOR";
})(IdeaAttributeColorNames = exports.IdeaAttributeColorNames || (exports.IdeaAttributeColorNames = {}));
var IdeaAttributeNames;
(function (IdeaAttributeNames) {
    IdeaAttributeNames["ANNOTATION_ATTRIBUTE_NAME_ATTRIBUTES"] = "ANNOTATION_ATTRIBUTE_NAME_ATTRIBUTES";
    IdeaAttributeNames["ANNOTATION_NAME_ATTRIBUTES"] = "ANNOTATION_NAME_ATTRIBUTES";
    IdeaAttributeNames["BAD_CHARACTER"] = "BAD_CHARACTER";
    IdeaAttributeNames["BOOKMARKS_ATTRIBUTES"] = "BOOKMARKS_ATTRIBUTES";
    IdeaAttributeNames["BREADCRUMBS_CURRENT"] = "BREADCRUMBS_CURRENT";
    IdeaAttributeNames["BREADCRUMBS_DEFAULT"] = "BREADCRUMBS_DEFAULT";
    IdeaAttributeNames["BREADCRUMBS_HOVERED"] = "BREADCRUMBS_HOVERED";
    IdeaAttributeNames["BREADCRUMBS_INACTIVE"] = "BREADCRUMBS_INACTIVE";
    IdeaAttributeNames["BREAKPOINT_ATTRIBUTES"] = "BREAKPOINT_ATTRIBUTES";
    IdeaAttributeNames["CONSOLE_BLUE_BRIGHT_OUTPUT"] = "CONSOLE_BLUE_BRIGHT_OUTPUT";
    IdeaAttributeNames["CONSOLE_BLUE_OUTPUT"] = "CONSOLE_BLUE_OUTPUT";
    IdeaAttributeNames["CONSOLE_CYAN_BRIGHT_OUTPUT"] = "CONSOLE_CYAN_BRIGHT_OUTPUT";
    IdeaAttributeNames["CONSOLE_CYAN_OUTPUT"] = "CONSOLE_CYAN_OUTPUT";
    IdeaAttributeNames["CONSOLE_ERROR_OUTPUT"] = "CONSOLE_ERROR_OUTPUT";
    IdeaAttributeNames["CONSOLE_GREEN_BRIGHT_OUTPUT"] = "CONSOLE_GREEN_BRIGHT_OUTPUT";
    IdeaAttributeNames["CONSOLE_GREEN_OUTPUT"] = "CONSOLE_GREEN_OUTPUT";
    IdeaAttributeNames["CONSOLE_MAGENTA_BRIGHT_OUTPUT"] = "CONSOLE_MAGENTA_BRIGHT_OUTPUT";
    IdeaAttributeNames["CONSOLE_MAGENTA_OUTPUT"] = "CONSOLE_MAGENTA_OUTPUT";
    IdeaAttributeNames["CONSOLE_NORMAL_OUTPUT"] = "CONSOLE_NORMAL_OUTPUT";
    IdeaAttributeNames["CONSOLE_RANGE_TO_EXECUTE"] = "CONSOLE_RANGE_TO_EXECUTE";
    IdeaAttributeNames["CONSOLE_RED_BRIGHT_OUTPUT"] = "CONSOLE_RED_BRIGHT_OUTPUT";
    IdeaAttributeNames["CONSOLE_RED_OUTPUT"] = "CONSOLE_RED_OUTPUT";
    IdeaAttributeNames["CONSOLE_SYSTEM_OUTPUT"] = "CONSOLE_SYSTEM_OUTPUT";
    IdeaAttributeNames["CONSOLE_USER_INPUT"] = "CONSOLE_USER_INPUT";
    IdeaAttributeNames["CONSOLE_YELLOW_BRIGHT_OUTPUT"] = "CONSOLE_YELLOW_BRIGHT_OUTPUT";
    IdeaAttributeNames["CONSOLE_YELLOW_OUTPUT"] = "CONSOLE_YELLOW_OUTPUT";
    IdeaAttributeNames["CSS_COLOR"] = "CSS.COLOR";
    IdeaAttributeNames["CSS_IMPORTANT"] = "CSS.IMPORTANT";
    IdeaAttributeNames["CSS_KEYWORD"] = "CSS.KEYWORD";
    IdeaAttributeNames["CSS_TAG_NAME"] = "CSS.TAG_NAME";
    IdeaAttributeNames["CSS_URL"] = "CSS.URL";
    IdeaAttributeNames["CTRL_CLICKABLE"] = "CTRL_CLICKABLE";
    IdeaAttributeNames["CUSTOM_KEYWORD1_ATTRIBUTES"] = "CUSTOM_KEYWORD1_ATTRIBUTES";
    IdeaAttributeNames["CUSTOM_KEYWORD2_ATTRIBUTES"] = "CUSTOM_KEYWORD2_ATTRIBUTES";
    IdeaAttributeNames["CUSTOM_KEYWORD3_ATTRIBUTES"] = "CUSTOM_KEYWORD3_ATTRIBUTES";
    IdeaAttributeNames["CUSTOM_KEYWORD4_ATTRIBUTES"] = "CUSTOM_KEYWORD4_ATTRIBUTES";
    IdeaAttributeNames["CUSTOM_STRING_ATTRIBUTES"] = "CUSTOM_STRING_ATTRIBUTES";
    IdeaAttributeNames["CUSTOM_VALID_STRING_ESCAPE_ATTRIBUTES"] = "CUSTOM_VALID_STRING_ESCAPE_ATTRIBUTES";
    IdeaAttributeNames["DEBUGGER_INLINED_VALUES"] = "DEBUGGER_INLINED_VALUES";
    IdeaAttributeNames["DEBUGGER_INLINED_VALUES_EXECUTION_LINE"] = "DEBUGGER_INLINED_VALUES_EXECUTION_LINE";
    IdeaAttributeNames["DEBUGGER_INLINED_VALUES_MODIFIED"] = "DEBUGGER_INLINED_VALUES_MODIFIED";
    IdeaAttributeNames["DEFAULT_BLOCK_COMMENT"] = "DEFAULT_BLOCK_COMMENT";
    IdeaAttributeNames["DEFAULT_CLASS_REFERENCE"] = "DEFAULT_CLASS_REFERENCE";
    IdeaAttributeNames["DEFAULT_COMMA"] = "DEFAULT_COMMA";
    IdeaAttributeNames["DEFAULT_CONSTANT"] = "DEFAULT_CONSTANT";
    IdeaAttributeNames["DEFAULT_DOC_COMMENT"] = "DEFAULT_DOC_COMMENT";
    IdeaAttributeNames["DEFAULT_DOC_COMMENT_TAG"] = "DEFAULT_DOC_COMMENT_TAG";
    IdeaAttributeNames["DEFAULT_DOC_COMMENT_TAG_VALUE"] = "DEFAULT_DOC_COMMENT_TAG_VALUE";
    IdeaAttributeNames["DEFAULT_DOC_MARKUP"] = "DEFAULT_DOC_MARKUP";
    IdeaAttributeNames["DEFAULT_FUNCTION_CALL"] = "DEFAULT_FUNCTION_CALL";
    IdeaAttributeNames["DEFAULT_FUNCTION_DECLARATION"] = "DEFAULT_FUNCTION_DECLARATION";
    IdeaAttributeNames["DEFAULT_IDENTIFIER"] = "DEFAULT_IDENTIFIER";
    IdeaAttributeNames["DEFAULT_INSTANCE_FIELD"] = "DEFAULT_INSTANCE_FIELD";
    IdeaAttributeNames["DEFAULT_INVALID_STRING_ESCAPE"] = "DEFAULT_INVALID_STRING_ESCAPE";
    IdeaAttributeNames["DEFAULT_KEYWORD"] = "DEFAULT_KEYWORD";
    IdeaAttributeNames["DEFAULT_LINE_COMMENT"] = "DEFAULT_LINE_COMMENT";
    IdeaAttributeNames["DEFAULT_METADATA"] = "DEFAULT_METADATA";
    IdeaAttributeNames["DEFAULT_NUMBER"] = "DEFAULT_NUMBER";
    IdeaAttributeNames["DEFAULT_OPERATION_SIGN"] = "DEFAULT_OPERATION_SIGN";
    IdeaAttributeNames["DEFAULT_REASSIGNED_LOCAL_VARIABLE"] = "DEFAULT_REASSIGNED_LOCAL_VARIABLE";
    IdeaAttributeNames["DEFAULT_REASSIGNED_PARAMETER"] = "DEFAULT_REASSIGNED_PARAMETER";
    IdeaAttributeNames["DEFAULT_SEMICOLON"] = "DEFAULT_SEMICOLON";
    IdeaAttributeNames["DEFAULT_STATIC_FIELD"] = "DEFAULT_STATIC_FIELD";
    IdeaAttributeNames["DEFAULT_STATIC_METHOD"] = "DEFAULT_STATIC_METHOD";
    IdeaAttributeNames["DEFAULT_STRING"] = "DEFAULT_STRING";
    IdeaAttributeNames["DEFAULT_TEMPLATE_LANGUAGE_COLOR"] = "DEFAULT_TEMPLATE_LANGUAGE_COLOR";
    IdeaAttributeNames["DEFAULT_VALID_STRING_ESCAPE"] = "DEFAULT_VALID_STRING_ESCAPE";
    IdeaAttributeNames["DELETED_TEXT_ATTRIBUTES"] = "DELETED_TEXT_ATTRIBUTES";
    IdeaAttributeNames["DIFF_CONFLICT"] = "DIFF_CONFLICT";
    IdeaAttributeNames["DIFF_DELETED"] = "DIFF_DELETED";
    IdeaAttributeNames["DIFF_INSERTED"] = "DIFF_INSERTED";
    IdeaAttributeNames["DIFF_MODIFIED"] = "DIFF_MODIFIED";
    IdeaAttributeNames["ERRORS_ATTRIBUTES"] = "ERRORS_ATTRIBUTES";
    IdeaAttributeNames["EVALUATED_EXPRESSION_ATTRIBUTES"] = "EVALUATED_EXPRESSION_ATTRIBUTES";
    IdeaAttributeNames["EXECUTIONPOINT_ATTRIBUTES"] = "EXECUTIONPOINT_ATTRIBUTES";
    IdeaAttributeNames["FOLDED_TEXT_ATTRIBUTES"] = "FOLDED_TEXT_ATTRIBUTES";
    IdeaAttributeNames["FOLLOWED_HYPERLINK_ATTRIBUTES"] = "FOLLOWED_HYPERLINK_ATTRIBUTES";
    IdeaAttributeNames["GHERKIN_REGEXP_PARAMETER"] = "GHERKIN_REGEXP_PARAMETER";
    IdeaAttributeNames["GRID_NULL_VALUE"] = "GRID_NULL_VALUE";
    IdeaAttributeNames["HAML_RUBY_CODE"] = "HAML_RUBY_CODE";
    IdeaAttributeNames["INJECTED_LANGUAGE_FRAGMENT"] = "INJECTED_LANGUAGE_FRAGMENT";
    IdeaAttributeNames["HTML_ATTRIBUTE_NAME"] = "HTML_ATTRIBUTE_NAME";
    IdeaAttributeNames["HTML_ATTRIBUTE_VALUE"] = "HTML_ATTRIBUTE_VALUE";
    IdeaAttributeNames["HTML_ENTITY_REFERENCE"] = "HTML_ENTITY_REFERENCE";
    IdeaAttributeNames["HTML_TAG"] = "HTML_TAG";
    IdeaAttributeNames["HTML_TAG_NAME"] = "HTML_TAG_NAME";
    IdeaAttributeNames["IDENTIFIER_UNDER_CARET_ATTRIBUTES"] = "IDENTIFIER_UNDER_CARET_ATTRIBUTES";
    IdeaAttributeNames["IMPLICIT_ANONYMOUS_CLASS_PARAMETER_ATTRIBUTES"] = "IMPLICIT_ANONYMOUS_CLASS_PARAMETER_ATTRIBUTES";
    IdeaAttributeNames["INFO_ATTRIBUTES"] = "INFO_ATTRIBUTES";
    IdeaAttributeNames["INLINE_PARAMETER_HINT"] = "INLINE_PARAMETER_HINT";
    IdeaAttributeNames["INLINE_PARAMETER_HINT_CURRENT"] = "INLINE_PARAMETER_HINT_CURRENT";
    IdeaAttributeNames["INLINE_PARAMETER_HINT_HIGHLIGHTED"] = "INLINE_PARAMETER_HINT_HIGHLIGHTED";
    IdeaAttributeNames["JAVA_KEYWORD"] = "JAVA_KEYWORD";
    IdeaAttributeNames["JSP_DIRECTIVE_NAME"] = "JSP_DIRECTIVE_NAME";
    IdeaAttributeNames["KOTLIN_ANNOTATION"] = "KOTLIN_ANNOTATION";
    IdeaAttributeNames["KOTLIN_BACKING_FIELD_VARIABLE"] = "KOTLIN_BACKING_FIELD_VARIABLE";
    IdeaAttributeNames["KOTLIN_DYNAMIC_FUNCTION_CALL"] = "KOTLIN_DYNAMIC_FUNCTION_CALL";
    IdeaAttributeNames["KOTLIN_DYNAMIC_PROPERTY_CALL"] = "KOTLIN_DYNAMIC_PROPERTY_CALL";
    IdeaAttributeNames["KOTLIN_LABEL"] = "KOTLIN_LABEL";
    IdeaAttributeNames["KOTLIN_NAMED_ARGUMENT"] = "KOTLIN_NAMED_ARGUMENT";
    IdeaAttributeNames["KOTLIN_SMART_CAST_RECEIVER"] = "KOTLIN_SMART_CAST_RECEIVER";
    IdeaAttributeNames["KOTLIN_SMART_CAST_VALUE"] = "KOTLIN_SMART_CAST_VALUE";
    IdeaAttributeNames["KOTLIN_SMART_CONSTANT"] = "KOTLIN_SMART_CONSTANT";
    IdeaAttributeNames["KOTLIN_TYPE_PARAMETER"] = "KOTLIN_TYPE_PARAMETER";
    IdeaAttributeNames["LESS_VARIABLE"] = "LESS_VARIABLE";
    IdeaAttributeNames["LINE_FULL_COVERAGE"] = "LINE_FULL_COVERAGE";
    IdeaAttributeNames["LINE_NONE_COVERAGE"] = "LINE_NONE_COVERAGE";
    IdeaAttributeNames["LINE_PARTIAL_COVERAGE"] = "LINE_PARTIAL_COVERAGE";
    IdeaAttributeNames["LIVE_TEMPLATE_ATTRIBUTES"] = "LIVE_TEMPLATE_ATTRIBUTES";
    IdeaAttributeNames["LOG_DEBUG_OUTPUT"] = "LOG_DEBUG_OUTPUT";
    IdeaAttributeNames["LOG_ERROR_OUTPUT"] = "LOG_ERROR_OUTPUT";
    IdeaAttributeNames["LOG_EXPIRED_ENTRY"] = "LOG_EXPIRED_ENTRY";
    IdeaAttributeNames["LOG_INFO_OUTPUT"] = "LOG_INFO_OUTPUT";
    IdeaAttributeNames["LOG_VERBOSE_OUTPUT"] = "LOG_VERBOSE_OUTPUT";
    IdeaAttributeNames["MARKED_FOR_REMOVAL_ATTRIBUTES"] = "MARKED_FOR_REMOVAL_ATTRIBUTES";
    IdeaAttributeNames["MATCHED_BRACE_ATTRIBUTES"] = "MATCHED_BRACE_ATTRIBUTES";
    IdeaAttributeNames["NOT_TOP_FRAME_ATTRIBUTES"] = "NOT_TOP_FRAME_ATTRIBUTES";
    IdeaAttributeNames["NOT_USED_ELEMENT_ATTRIBUTES"] = "NOT_USED_ELEMENT_ATTRIBUTES";
    IdeaAttributeNames["PROPERTIES_INVALID_STRING_ESCAPE"] = "PROPERTIES.INVALID_STRING_ESCAPE";
    IdeaAttributeNames["REGEXP_CHAR_CLASS"] = "REGEXP.CHAR_CLASS";
    IdeaAttributeNames["REGEXP_META"] = "REGEXP.META";
    IdeaAttributeNames["REGEXP_QUOTE_CHARACTER"] = "REGEXP.QUOTE_CHARACTER";
    IdeaAttributeNames["SASS_MIXIN"] = "SASS_MIXIN";
    IdeaAttributeNames["SASS_VARIABLE"] = "SASS_VARIABLE";
    IdeaAttributeNames["SEARCH_RESULT_ATTRIBUTES"] = "SEARCH_RESULT_ATTRIBUTES";
    IdeaAttributeNames["SPY_JS_EXCEPTION"] = "SPY-JS.EXCEPTION";
    IdeaAttributeNames["SPY_JS_EXECUTION_TIME"] = "SPY-JS.EXECUTION_TIME";
    IdeaAttributeNames["SPY_JS_FUNCTION_SCOPE"] = "SPY-JS.FUNCTION_SCOPE";
    IdeaAttributeNames["SPY_JS_PATH_LEVEL_ONE"] = "SPY-JS.PATH_LEVEL_ONE";
    IdeaAttributeNames["SPY_JS_PATH_LEVEL_TWO"] = "SPY-JS.PATH_LEVEL_TWO";
    IdeaAttributeNames["SPY_JS_PROGRAM_SCOPE"] = "SPY-JS.PROGRAM_SCOPE";
    IdeaAttributeNames["STATIC_METHOD_ACCESS"] = "Static method access";
    IdeaAttributeNames["STATIC_PROPERTY_REFERENCE_ID"] = "Static property reference ID";
    IdeaAttributeNames["TAPESTRY_COMPONENT_TAG"] = "TAPESTRY_COMPONENT_TAG";
    IdeaAttributeNames["TEMPLATE_VARIABLE_ATTRIBUTES"] = "TEMPLATE_VARIABLE_ATTRIBUTES";
    IdeaAttributeNames["TEXT"] = "TEXT";
    IdeaAttributeNames["TEXT_SEARCH_RESULT_ATTRIBUTES"] = "TEXT_SEARCH_RESULT_ATTRIBUTES";
    IdeaAttributeNames["TODO_DEFAULT_ATTRIBUTES"] = "TODO_DEFAULT_ATTRIBUTES";
    IdeaAttributeNames["TYPE_PARAMETER_NAME_ATTRIBUTES"] = "TYPE_PARAMETER_NAME_ATTRIBUTES";
    IdeaAttributeNames["UNMATCHED_BRACE_ATTRIBUTES"] = "UNMATCHED_BRACE_ATTRIBUTES";
    IdeaAttributeNames["VELOCITY_DIRECTIVE"] = "VELOCITY_DIRECTIVE";
    IdeaAttributeNames["VELOCITY_KEYWORD"] = "VELOCITY_KEYWORD";
    IdeaAttributeNames["VELOCITY_SCRIPTING_BACKGROUND"] = "VELOCITY_SCRIPTING_BACKGROUND";
    IdeaAttributeNames["WARNING_ATTRIBUTES"] = "WARNING_ATTRIBUTES";
    IdeaAttributeNames["WRITE_IDENTIFIER_UNDER_CARET_ATTRIBUTES"] = "WRITE_IDENTIFIER_UNDER_CARET_ATTRIBUTES";
    IdeaAttributeNames["WRITE_SEARCH_RESULT_ATTRIBUTES"] = "WRITE_SEARCH_RESULT_ATTRIBUTES";
    IdeaAttributeNames["WRONG_REFERENCES_ATTRIBUTES"] = "WRONG_REFERENCES_ATTRIBUTES";
    IdeaAttributeNames["XML_ATTRIBUTE_NAME"] = "XML_ATTRIBUTE_NAME";
    IdeaAttributeNames["XML_ENTITY_REFERENCE"] = "XML_ENTITY_REFERENCE";
    IdeaAttributeNames["XML_PROLOGUE"] = "XML_PROLOGUE";
    IdeaAttributeNames["XML_TAG"] = "XML_TAG";
    IdeaAttributeNames["XML_TAG_NAME"] = "XML_TAG_NAME";
    IdeaAttributeNames["XPATH_KEYWORD"] = "XPATH.KEYWORD";
    IdeaAttributeNames["XPATH_XPATH_NAME"] = "XPATH.XPATH_NAME";
    IdeaAttributeNames["YAML_ANCHOR"] = "YAML_ANCHOR";
})(IdeaAttributeNames = exports.IdeaAttributeNames || (exports.IdeaAttributeNames = {}));
