// code-theme-converter/dist/types/sublime.js
"use strict";
/* eslint-disable @typescript-eslint/camelcase */
Object.defineProperty(exports, "__esModule", { value: true });
var SublimeUIClass;
(function (SublimeUIClass) {
    SublimeUIClass["title_bar"] = "title_bar";
    SublimeUIClass["sidebar_heading"] = "sidebar_heading";
    SublimeUIClass["sidebar_container"] = "sidebar_container";
    SublimeUIClass["sidebar_tree"] = "sidebar_tree";
    SublimeUIClass["sidebar_label"] = "sidebar_label";
    SublimeUIClass["tree_row"] = "tree_row";
    SublimeUIClass["window"] = "window";
    SublimeUIClass["close_button"] = "close_button";
    SublimeUIClass["disclosure_button_control"] = "disclosure_button_control";
    SublimeUIClass["icon_folder"] = "icon_folder";
    SublimeUIClass["icon_folder_loading"] = "icon_folder_loading";
    SublimeUIClass["icon_folder_dup"] = "icon_folder_dup";
    SublimeUIClass["icon_file_type"] = "icon_file_type";
    SublimeUIClass["file_system_entry"] = "file_system_entry";
    SublimeUIClass["scroll_area_control"] = "scroll_area_control";
    SublimeUIClass["vcs_status_badge"] = "vcs_status_badge";
    SublimeUIClass["vcs_changes_annotation"] = "vcs_changes_annotation";
    SublimeUIClass["vcs_branch_icon"] = "vcs_branch_icon";
    SublimeUIClass["vcs_status"] = "vcs_status";
    SublimeUIClass["label_control"] = "label_control";
    SublimeUIClass["status_button"] = "status_button";
    SublimeUIClass["status_container"] = "status_container";
    SublimeUIClass["sidebar_button_control"] = "sidebar_button_control";
    SublimeUIClass["status_bar"] = "status_bar";
    SublimeUIClass["tool_tip_label_control"] = "tool_tip_label_control";
    SublimeUIClass["tool_tip_control"] = "tool_tip_control";
    SublimeUIClass["title_label_control"] = "title_label_control";
    SublimeUIClass["icon_use_gitignore"] = "icon_use_gitignore";
    SublimeUIClass["icon_button_control"] = "icon_button_control";
    SublimeUIClass["icon_use_buffer"] = "icon_use_buffer";
    SublimeUIClass["icon_context"] = "icon_context";
    SublimeUIClass["icon_preserve_case"] = "icon_preserve_case";
    SublimeUIClass["icon_highlight"] = "icon_highlight";
    SublimeUIClass["icon_in_selection"] = "icon_in_selection";
    SublimeUIClass["icon_wrap"] = "icon_wrap";
    SublimeUIClass["icon_whole_word"] = "icon_whole_word";
    SublimeUIClass["icon_case"] = "icon_case";
    SublimeUIClass["icon_regex"] = "icon_regex";
    SublimeUIClass["icon_button_group"] = "icon_button_group";
    SublimeUIClass["button_control"] = "button_control";
    SublimeUIClass["dropdown_button_control"] = "dropdown_button_control";
    SublimeUIClass["text_line_control"] = "text_line_control";
    SublimeUIClass["scroll_bar_control"] = "scroll_bar_control";
    SublimeUIClass["overlay_control"] = "overlay_control";
    SublimeUIClass["puck_control"] = "puck_control";
    SublimeUIClass["scroll_corner_control"] = "scroll_corner_control";
    SublimeUIClass["scroll_track_control"] = "scroll_track_control";
    SublimeUIClass["progress_gauge_control"] = "progress_gauge_control";
    SublimeUIClass["progress_bar_control"] = "progress_bar_control";
    SublimeUIClass["dialog"] = "dialog";
    SublimeUIClass["panel_close_button"] = "panel_close_button";
    SublimeUIClass["panel_grid_control"] = "panel_grid_control";
    SublimeUIClass["panel_control"] = "panel_control";
    SublimeUIClass["tabset_control"] = "tabset_control";
    SublimeUIClass["tab_control"] = "tab_control";
    SublimeUIClass["tab_label"] = "tab_label";
    SublimeUIClass["tab_close_button"] = "tab_close_button";
    SublimeUIClass["scroll_tabs_left_button"] = "scroll_tabs_left_button";
    SublimeUIClass["scroll_tabs_right_button"] = "scroll_tabs_right_button";
    SublimeUIClass["show_tabs_dropdown_button"] = "show_tabs_dropdown_button";
    SublimeUIClass["quick_panel"] = "quick_panel";
    SublimeUIClass["mini_quick_panel_row"] = "mini_quick_panel_row";
    SublimeUIClass["quick_panel_row"] = "quick_panel_row";
    SublimeUIClass["quick_panel_entry"] = "quick_panel_entry";
    SublimeUIClass["quick_panel_label"] = "quick_panel_label";
    SublimeUIClass["quick_panel_path_label"] = "quick_panel_path_label";
    SublimeUIClass["quick_panel_label_hint"] = "quick_panel_label hint";
    SublimeUIClass["quick_panel_label_key_binding"] = "quick_panel_label key_binding";
    SublimeUIClass["grid_layout_control"] = "grid_layout_control";
})(SublimeUIClass = exports.SublimeUIClass || (exports.SublimeUIClass = {}));
var SublimeUIAttributes;
(function (SublimeUIAttributes) {
    SublimeUIAttributes["file_dark"] = "file_dark";
    SublimeUIAttributes["file_medium_dark"] = "file_medium_dark";
    SublimeUIAttributes["file_medium"] = "file_medium";
    SublimeUIAttributes["file_light"] = "file_light";
    SublimeUIAttributes["hover"] = "hover";
    SublimeUIAttributes["not_hover"] = "!hover";
    SublimeUIAttributes["selectable"] = "selectable";
    SublimeUIAttributes["expandable"] = "expandable";
    SublimeUIAttributes["scrollable"] = "scrollable";
    SublimeUIAttributes["dirty"] = "dirty";
    SublimeUIAttributes["expanded"] = "expanded";
    SublimeUIAttributes["modified"] = "modified";
    SublimeUIAttributes["missing"] = "missing";
    SublimeUIAttributes["staged"] = "staged";
    SublimeUIAttributes["added"] = "added";
    SublimeUIAttributes["deleted"] = "deleted";
    SublimeUIAttributes["unmerged"] = "unmerged";
    SublimeUIAttributes["ignored"] = "ignored";
    SublimeUIAttributes["untracked"] = "untracked";
    SublimeUIAttributes["background"] = "background";
    SublimeUIAttributes["foreground"] = "foreground";
    SublimeUIAttributes["selected"] = "selected";
    SublimeUIAttributes["backgroSublimeUIAttributes"] = "backgroSublimeUIAttributes";
    SublimeUIAttributes["transient"] = "transient";
    SublimeUIAttributes["not_selected"] = "!selected";
})(SublimeUIAttributes = exports.SublimeUIAttributes || (exports.SublimeUIAttributes = {}));
var SublimeUISettings;
(function (SublimeUISettings) {
    SublimeUISettings["SYSTEM"] = "system";
    SublimeUISettings["BOLD_FOLDER_LABELS"] = "bold_folder_labels";
    SublimeUISettings["overlay_scroll_bars"] = "overlay_scroll_bars";
    SublimeUISettings["not_overlay_scroll_bars"] = "!overlay_scroll_bars";
    SublimeUISettings["dark"] = "dark";
    SublimeUISettings["light"] = "light";
    SublimeUISettings["bold_folder_labels"] = "bold_folder_labels";
    SublimeUISettings["mouse_wheel_switches_tabs"] = "mouse_wheel_switches_tabs";
    SublimeUISettings["NOT_ENABLE_TAB_SCROLLING"] = "!enable_tab_scrolling";
    SublimeUISettings["highlight_modified_tabs"] = "highlight_modified_tabs";
    SublimeUISettings["show_tab_close_buttons_on_left"] = "show_tab_close_buttons_on_left";
    SublimeUISettings["not_show_tab_close_buttons"] = "show_tab_close_buttons";
    SublimeUISettings["show_tab_close_buttons"] = "show_tab_close_buttons";
})(SublimeUISettings = exports.SublimeUISettings || (exports.SublimeUISettings = {}));
var Platform;
(function (Platform) {
    Platform["WINDOWS"] = "windows";
})(Platform = exports.Platform || (exports.Platform = {}));
