// code-theme-converter/dist/idea/plugin.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const js2xmlparser_1 = require("js2xmlparser");
function createPluginXml({ plugin, vendor, theme }) {
    return js2xmlparser_1.parse('idea-plugin', {
        id: {
            '#': plugin.id
        },
        name: {
            '#': plugin.name
        },
        version: {
            '#': plugin.version
        },
        vendor: {
            '@': {
                email: vendor.email,
                url: vendor.url
            },
            '#': vendor.name
        },
        description: {
            '#': `<![CDATA[${plugin.description}]]>`
        },
        'change-notes': {
            '#': `<![CDATA[${plugin.changelog}]]>`
        },
        'idea-version': {
            '@': {
                'since-build': '191.0'
            }
        },
        depends: {
            '#': 'com.intellij.modules.lang'
        },
        extensions: {
            '@': {
                defaultExtensionNs: 'com.intellij'
            },
            themeProvider: {
                '@': {
                    id: plugin.id,
                    path: `/${theme.name}.theme.json`
                }
            }
        },
        actions: {}
    });
}
exports.createPluginXml = createPluginXml;
